File: gui/src/Rectangle.cpp
// NebulaOS GUI - Rectangle Implementation
// =======================================
//
// Implementation of Rectangle class

#include "../include/Rectangle.h"
#include "../include/GuiTypes.h"
#include "../../lib/include/string.h"
#include "../../kernel/common/include/nebula.h"

namespace NebulaOS {
namespace GUI {

// -----------------------------------------------------------------------------
// ToString - Convert rectangle to string representation
// -----------------------------------------------------------------------------
void Rectangle::ToString(char* buffer, size_t size) const {
    if (!buffer || size == 0) return;
    
    snprintf(buffer, size, "Rectangle(%d, %d, %d, %d)", x, y, width, height);
}

} // namespace GUI
} // namespace NebulaOS

File: gui/src/rectangle.rs
pub struct Rectangle {
    pub x: i32,
    pub y: i32,
    pub width: u32,
    pub height: u32,
}

impl Rectangle {
    pub const fn new(x: i32, y: i32, width: u32, height: u32) -> Self {
        Rectangle { x, y, width, height }
    }
}
